File: frontend-register-login/src/interfaces/community.ts
// export interface Community {
//     id:number;
//     name:string;
//     description?:string;
//     creationDate:string;
// }

import { Post } from "./post";

export interface Community {
    id: number;
    name: string;
    description?: string;
    creationDate: string;
    moderators: any | null; 
    subscribers: Subscriber[]; 
    posts: Post; 
    communityInfo:string|null;

}
  export interface Subscriber {
    id: number;
    username: string;
    password: string;
    email: string;
    moderator: any | null; 
    posts: any |null; 
    comments: any; 
    dateOfAccountCreated: string;
  }

  export interface CommunityToSend{

    name?:string,
    description?:string
  }

File: WebTemplate/Models/Community.cs
using System.Text.Json.Serialization;

namespace WebTemplate.Models;


public class Community{

    [Key]
    public int Id { get; set; }

    [MaxLength(50)]
    public required string Name { get; set; }

    [MaxLength(20000)]
    public string? Description { get; set; }

    public DateTime CreationDate { get; set; }=DateTime.Now;

    public List<User>? Moderators { get; set; }


    public List<User>? Subscribers { get; set; }

    public List<Post>? Posts { get; set; }


    public string? CommInfo { get; set; }
}